sdk/azcore/response.go

```diff
@@ -26,15 +26,16 @@ type Response struct {
 	*http.Response
 }
 
-func (r *Response) payload() []byte {
-	if r.Body == nil {
-		return nil
-	}
+func (r *Response) payload() ([]byte, error) {
 	// r.Body won't be a nopClosingBytesReader if downloading was skipped
 	if buf, ok := r.Body.(*nopClosingBytesReader); ok {
-		return buf.Bytes()
+		return buf.Bytes(), nil
 	}
-	return nil
+	bytesBody, err := ioutil.ReadAll(r.Body)
```

Comment:
need to call `r.Body.Close()` after this line

Reply:
Done. I didnt check the error for this, let me know if I should.

Reply:
No need to check the error, calling `Close()` on an `io.ReadCloser` is one of the rare patterns where we don't need to check it.